app/Http/Controllers/Auth/VATSIMOAuthController.php

```diff
@@ -0,0 +1,49 @@
+<?php
+
+namespace App\Http\Controllers\Auth;
+
+use League\OAuth2\Client\Provider\Exception\IdentityProviderException;
+use League\OAuth2\Client\Provider\GenericProvider;
+
+class VATSIMOAuthController extends GenericProvider
+{
+    /**
+     * @var GenericProvider
+     */
+    private $provider;
+
+    /**
+     * Initializes the provider variable.
+     */
+    public function __construct()
+    {
+        parent::__construct([
+            'clientId'                => config('vatsim-connect.id'),
+            'clientSecret'            => config('vatsim-connect.secret'),
+            'redirectUri'             => route('login.post'),
+            'urlAuthorize'            => config('vatsim-connect.base').'/oauth/authorize',
+            'urlAccessToken'          => config('vatsim-connect.base').'/oauth/token',
+            'urlResourceOwnerDetails' => config('vatsim-connect.base').'/api/user',
+            'scopes'                  => config('vatsim-connect.scopes'),
+            'scopeSeparator'          => ' ',
```

Comment:
Just checking this is intended to be an empty string with a bit of whitespace (length of string is still technically zero since the `\0` is not counted.

Reply:
This is from the example repo and separates the config out `'scopes' => explode(',', env('VATSIM_OAUTH_SCOPES')),`